cpp/daal/src/threading/threading.cpp

```diff
@@ -42,6 +42,9 @@
 
 using namespace daal::services;
 
+static std::shared_ptr<tbb::task_scheduler_handle> globalSchedulerHandle = nullptr;
```

Comment:
Why do we need static shared_ptr if we make sure that the object stored in it is deleted before the deletion of smart pointer? Could we use a normal pointer instead?